core/src/test/scala/unit/kafka/utils/TestUtils.scala

```diff
@@ -1124,24 +933,18 @@ object TestUtils extends Logging {
     }, msg = msg, pause = 0L, waitTimeMs = waitTimeMs)
   }
 
-  /**
-   * Wait for the presence of an optional value.
-   *
-   * @param func The function defining the optional value
-   * @param msg Error message in the case that the value never appears
-   * @param waitTimeMs Maximum time to wait
-   * @return The unwrapped value returned by the function
-   */
-  def awaitValue[T](func: () => Option[T], msg: => String, waitTimeMs: Long = JTestUtils.DEFAULT_MAX_WAIT_MS): T = {
-    var value: Option[T] = None
-    waitUntilTrue(() => {
-      value = func()
-      value.isDefined
-    }, msg, waitTimeMs)
-    value.get
+  def subscribeAndWaitForRecords(topic: String,
```

Comment:
why adding this new method? it is unused.

Reply:
`assertBadConfigContainingMessage` this method are used by two class, so I didn't move it. 
The `kafka.KafkaConfigTest` and `server.KafkaConfigTest` are using it

Reply:
Other methods refactored by your comment.